src/main/java/uk/gov/verifiablelog/merkletree/InMemoryPowOfTwo.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.verifiablelog.merkletree;
+
+
+import java.util.TreeMap;
+
+public class InMemoryPowOfTwo implements MemoizationStore {
+
+    private TreeMap<Integer, TreeMap<Integer, byte[]>> data;
+
+    public InMemoryPowOfTwo() {
+        data = new TreeMap<>();
+    }
+
+
+    @Override
+    public void put(Integer start, Integer size, byte[] value) {
+        if (((start == 0) || (Integer.bitCount(start) == 1)) &&
+                ((size == 0) || (Integer.bitCount(size) == 1))) {
```

Comment:
also we should (at some point) test this class. shouldn't block merging this PR though.